ampd/src/solana/json_rpc.rs

```diff
@@ -0,0 +1,126 @@
+use async_trait::async_trait;
+use ethers::providers::{JsonRpcClient, ProviderError};
+use mockall::automock;
+use serde::{Deserialize, Serialize};
+use solana_account_decoder::UiAccountEncoding;
+use solana_sdk::commitment_config::{CommitmentConfig, CommitmentLevel};
+// use solana_sdk::transaction::Transaction;
+
+use crate::json_rpc::Client;
+
+type Result<T> = error_stack::Result<T, ProviderError>;
+
+// TODO: This should come from solana-sdk crate
```

Comment:
I'm a bit confused, but I think you refactored this another PR to not use these copy pasted types 🤔 
If so, we should make sure that the refactored code gets merged and not this one, correct?

Reply:
And it seems to be the other way around for tests. 
You have tests here, but not in the other PR, if I'm not mistaken.

Reply:
Now this PR is in sync with the `solana` branch, which contains the needed changes in order to use the solana crates types.